### DISCUSSION TITLE
Gameboy dump issue

### DISCUSSION THREAD

**Discussion by drpaneas** [sanni/cartreader#297] [Need help]
I've just got mine (pre-assembled) but any Gameboy cartridge I try to dump is not recognized. No name, no rom, just some broken files. How can I troubleshoot what's wrong there?

**Comment by sanni** [comment]
The most efficient way would be to contact your seller and ask for a replacement but if that is not an option or if you like the challenge of fixing it yourself then I found that the fastest way to troubleshoot the hardware is to compare a broken dump to a known good one of the same game in an hex editor and then try to see a pattern which points to a specific address or data line being the culprit. 

It's important to know that with the hardware V3 all slots are connected in parallel so the hardware defect could be on any of the slots and still have an effect on the Gameboy slot. This is something I changed with hardware V4.

To find the issue you should use a standard Game Boy (Color) game, not Game Boy Advance since GB(C) has an easier protocol where address lines and data lines are split.

Here is an example of this way to troubleshoot from the perspective of the N64 slot: https://forum.arduino.cc/t/rom-reader-for-super-nintendo-super-famicom-game-cartridges/155260/1270?u=sanni

And here the same issue but seen from the GB slot: https://forum.arduino.cc/t/rom-reader-for-super-nintendo-super-famicom-game-cartridges/155260/1272?u=sanni

So using different slot you can also crosscheck the issue. In this case the user had a bad solder connection on Arduino Pin A8.

**Comment by Dakkaron** [comment]
Looks like the kickstarter mess is rolling in... If you can't get the device fixed by the seller, try to resolder every single solder connection on the board.

**Reply by SAVE-THE-HERO** [reply]
Hallo Dakkaron.

I test before I ship, 
But If buyer have any problems with any of the devices I sell, I will take full responsibility.
Please don't worry, I will inform the buyer about it.

However, I am not very knowledgeable about programming, so if I have any questions, I will guide you to the forum.

Like this.
https://twitter.com/akikawa134/status/1466913896397422592?s=20
So If you can, I would be happy to you answer such questions.

With all my gratitude

**Reply by Dakkaron** [reply]
To be honest, no, I am not really happy about that.

It really sucks if a seller has no idea about the product they are selling. Get to know the product, learn some programming and understand what you are selling. That is your responsibility.

You started it, you clean it up.

**Reply by SAVE-THE-HERO** [reply]
Yes I know you think about it,
But I don't think it's worth arguing about any more.
It's no different to a child's quarrel.
I also believe that the only way to get results is to see them.

With all my gratitude

**Reply by Androxilogin** [reply]
> To be honest, no, I am not really happy about that.
> 
> It really sucks if a seller has no idea about the product they are selling. Get to know the product, learn some programming and understand what you are selling. That is your responsibility.
> 
> You started it, you clean it up.


I agree completely. In this instance, I got the notification after the buyer had already reached out here. I try to shield my own issues from making their way here. Especially when I am the one at fault.

**Reply by SAVE-THE-HERO** [reply]
Yes I think too.
Thank you  Androxilogin.

With all my gratitude

**Comment by Androxilogin** [comment]
> Looks like the kickstarter mess is rolling in... If you can't get the device fixed by the seller, try to resolder every single solder connection on the board.

Unfortunately this is my problem. I used Kester solder rosin core. The flux apparently backfed into the switches and hardened even after vigorously cleaning the board and components afterwards. So it was working fine while in my hands then presented itself later on. We've been in contact about this issue and walking through it.

**Comment by sanni** [comment]
I remember having a similar problem once. One of the switches measured a resistance higher than 1 Ohm between the actual pins, so not causes by a cold joint, but heating it up still fixed it and the resistance did go down again. I concluded that it might have been oxidation or something like that but it could also have been flux or some IPA between the mechanical parts inside the switch.
